src/components/query-selector/measure-file-upload.tsx

```diff
@@ -0,0 +1,130 @@
+import { measureBundleState } from '@/state/measure-bundle';
+import { Center, Text } from '@mantine/core';
+import { Dropzone } from '@mantine/dropzone';
+import { showNotification } from '@mantine/notifications';
+import { IconAlertCircle, IconFileCheck, IconFileImport } from '@tabler/icons-react';
+import { bulkQueries, group } from 'fqm-bulk-utils';
+import { useState } from 'react';
+import { useRecoilState } from 'recoil';
+
+export default function MeasureFileUpload() {
+  const [dropStatus, setDropStatus] = useState<'idle' | 'accept' | 'reject'>('idle');
+  const [measureBundle, setMeasureBundle] = useRecoilState(measureBundleState);
+
+  const rejectUpload = (message: string) => {
+    setDropStatus('reject');
+    showNotification({
+      icon: <IconAlertCircle />,
+      title: 'Bundle upload failed',
+      message: `There was an issue uploading your bundle: ${message}`,
+      color: 'red'
+    });
+  };
+
+  const handleFileRead = (file: File) => {
+    const reader = new FileReader();
+    reader.onload = () => {
+      try {
+        const bundle = JSON.parse(reader.result as string) as fhir4.Bundle;
+
+        const measures = bundle.entry?.filter(r => r?.resource?.resourceType === 'Measure');
+        if (bundle.resourceType !== 'Bundle') {
+          rejectUpload('Uploaded file must be a JSON FHIR Resource of type "Bundle"');
+          return;
+        } else if (!measures || measures.length !== 1) {
+          rejectUpload('Uploaded bundle must contain exactly one resource of type "Measure"');
+          return;
+        }
+
+        handleBundleProcessing(bundle, file.name);
+      } catch (error) {
+        setDropStatus('reject');
+        showNotification({
+          id: 'invalid-json',
+          icon: <IconAlertCircle />,
+          title: 'Invalid JSON',
+          message: 'The uploaded file is not valid JSON.',
+          color: 'red'
+        });
+      }
+    };
+    reader.readAsText(file);
+  };
+
+  const handleBundleProcessing = async (bundle: fhir4.Bundle, fileName: string) => {
+    let text = 'No group content';
+    let queries = '';
+
+    try {
```

Comment:
A bit of an implementation question here. Do we want to be able to move to the next page if either of these calls to fqm-bulk-utils fails? 
Right now group and bulkQueries are sort of dependent on each other because they're in the same try block. 
Theoretically one of these calls (the second one) could fail and you could still use the results of the other call on the next page. What if we separate the calls into separate try blocks so that if one of either call fails, you could still use the results of the other?
Or we could disallow navigation to the next page in the case of either of them failing because the assumption is that something is wonky.

Reply:
What is groupText used for in this case?